editor/src/components/canvas/commands/push-intended-bounds-and-update-hugging-elements-command.ts

```diff
@@ -0,0 +1,177 @@
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import * as EP from '../../../core/shared/element-path'
+import type { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import type { ElementPath } from '../../../core/shared/project-file-types'
+import * as PP from '../../../core/shared/property-path'
+import type { EditorState, EditorStatePatch } from '../../editor/store/editor-state'
+import { cssPixelLength, type FlexDirection } from '../../inspector/common/css-utils'
+import {
+  flexChildAndBottomRightProps,
+  isHuggingParent,
+  prunePropsCommands,
+} from '../../inspector/inspector-common'
+import type { CanvasFrameAndTarget } from '../canvas-types'
+import type { BaseCommand, CanvasCommand, CommandFunctionResult } from './commands'
+import { foldAndApplyCommandsSimple } from './commands'
+import { setCssLengthProperty, setExplicitCssValue } from './set-css-length-command'
+import { setProperty } from './set-property-command'
+import { showToastCommand } from './show-toast-command'
+
+export interface PushIntendedBoundsAndUpdateHuggingElements extends BaseCommand {
+  type: 'PUSH_INTENDED_BOUNDS_AND_UPDATE_HUGGING_ELEMENTS'
+  value: Array<CanvasFrameAndTarget>
+}
+
+export function pushIntendedBoundsAndUpdateHuggingElements(
+  value: Array<CanvasFrameAndTarget>,
+): PushIntendedBoundsAndUpdateHuggingElements {
+  return {
+    type: 'PUSH_INTENDED_BOUNDS_AND_UPDATE_HUGGING_ELEMENTS',
+    whenToRun: 'always',
+    value: value,
+  }
+}
+
+export const runPushIntendedBoundsAndUpdateHuggingElements = (
+  editor: EditorState,
+  command: PushIntendedBoundsAndUpdateHuggingElements,
+): CommandFunctionResult => {
+  const { updatedEditor: editorAfterHuggingElements } = getUpdateResizeHuggingElementsCommands(
+    editor,
+    command,
+  )
+
+  let editorStatePatches: Array<EditorStatePatch> = [
+    {
+      projectContents: {
+        $set: editorAfterHuggingElements.projectContents,
+      },
+      toasts: {
+        $set: editorAfterHuggingElements.toasts,
+      },
+    },
+  ]
+
+  return {
+    editorStatePatches: editorStatePatches,
+    commandDescription: `Set Intended Bounds for hugging elements ${command.value
+      .map((c) => EP.toString(c.target))
+      .join(', ')}`,
+  }
+}
+
+function getHuggingElementContentsStatus(
+  jsxMetadata: ElementInstanceMetadataMap,
+  path: ElementPath,
+): 'empty' | 'contains-only-absolute' | 'contains-some-absolute' | 'non-empty' {
+  const children = MetadataUtils.getChildrenUnordered(jsxMetadata, path)
+  const absoluteChildren = children.filter(MetadataUtils.isPositionAbsolute).length
+  if (children.length === 0) {
+    return 'empty'
+  } else if (absoluteChildren === children.length) {
+    return 'contains-only-absolute'
+  } else if (absoluteChildren > 0) {
+    return 'contains-some-absolute'
+  } else {
+    return 'non-empty'
+  }
+}
+
+function getUpdateResizeHuggingElementsCommands(
```

Comment:
This function should be renamed, because it's not just getting the commands, it is getting them and then applying them, and returning the result of that

Reply:
[`6ea21c1` (#4359)](https://github.com/concrete-utopia/utopia/pull/4359/commits/6ea21c1c1b3b7bb9691ae2e869db41f01d63fa59)